public/app/core/components/TimelineChart/timeline.ts

```diff
@@ -209,7 +220,7 @@ export function getConfig(opts: TimelineCoreOptions) {
             for (let ix = 0; ix < dataY.length; ix++) {
               let yVal = dataY[ix];
 
-              if (yVal != null || mappedNull) {
+              if (shouldDrawBox(sidx, yVal)) {
```

Comment:
tldr, the bug was primarily located at the three spots in code we checked `yVal != null || mappedNull`. It didn't account for the `NaN` cases, nor did it account for if `null` was present but NullAndNaN mapping was enabled. we opted to just reorganize things a bit, mostly by accounting for `NaN` and `null` separately.